feat: Make automatic screenshot taking configurable

mrmacete opened the pull request:
- disable by default
- initial disabling avoidable by setting env var DISABLE_SCREENSHOTS=NO
- overridable on session start via `disableAutomaticScreenshots` capability

The reason for this change is that sometimes it's useful to turn this feature back on for debugging purposes, especially when working on complex setups with remote devices.

mrmacete committed:
feat: Make automatic screenshot taking configurable

- disable by default
- initial disabling avoidable by setting env var DISABLE_SCREENSHOTS=NO
- overridable on session start via capability

mykola-mokhnach commented on WebDriverAgentLib/Utilities/FBConfiguration.m:
```diff
@@ -89,6 +89,25 @@ + (void)disableScreenshots
++ (void)disableScreenshotsUnlessEnvironment
+{
+  BOOL disable = YES;
+  NSString * envSetting = NSProcessInfo.processInfo.environment[@"DISABLE_SCREENSHOTS"];
```
This might be confusing. How about changing the variable name to `ENABLE_SCREENSHOTS` and actually enable them if this variable is defined without checking its actual value?

mykola-mokhnach replied:
I would also rename the env variable itself to make its name more descriptive: ENABLE_AUTOMATIC_SCREENSHOTS

mrmacete committed:
Rename env variable and refactor env check

So that the mere presence of ENABLE_AUTOMATIC_SCREENSHOTS in the env is
enough to enable them, regardless of the env var value.

mykola-mokhnach commented on WebDriverAgentLib/Utilities/FBConfiguration.h:
```diff
@@ -43,9 +43,15 @@ extern NSString *const FBSnapshotMaxDepthKey;
+/*! Enables XCTest automated screenshots taking */
++ (void)enableScreenshots;
+
+/*! Disables XCTest automated screenshots taking, unless DISABLE_SCREENSHOTS env variable is set to NO */
```
Please update the docstring

mykola-mokhnach commented on WebDriverAgentLib/Utilities/FBConfiguration.m:
```diff
@@ -89,6 +89,20 @@ + (void)disableScreenshots
+  [[NSUserDefaults standardUserDefaults] setBool:NO forKey:@"DisableScreenshots"];
+}
+
++ (void)disableScreenshotsUnlessEnvironment
```
I think it makes sense to inline this logic into UITestingUITests, because it is only used there

mrmacete replied:
right! thanks

mrmacete committed:
Inline screenshot enabling logic

mykola-mokhnach commented on WebDriverAgentLib/Commands/FBSessionCommands.m:
```diff
@@ -97,6 +97,14 @@ + (NSArray *)routes
     [FBConfiguration setShouldUseSingletonTestManager:[requirements[@"shouldUseSingletonTestManager"] boolValue]];
   }
+  if (requirements[@"disableAutomaticScreenshots"]) {
+    BOOL disable = [requirements[@"disableAutomaticScreenshots"] boolValue];
```
this intermediate variable is not needed

mrmacete committed:
Remove redundant intermediate variable
